### PR TITLE
add pericardium model boundary force

### LINKED ISSUE #78: add dashpots
@marshallrdavey want to give this a try? Now that we have normal vectors, velocities, and displacements it should be straightforward to add this as a new boundary force.

### PR DESCRIPTION
**Pull request by marshallrdavey**
fixes #78 

Does this works the way I think it does where `normal_vector` and `get_velocity_values` are computed at the quadrature points on the face?

### CODE REVIEW

File: source/mechanics/force_contribution_lib.cc [hunk 1]
```diff
@@ -387,6 +387,155 @@ namespace fdl
...
+        extractor.get_function_values_from_local_dof_values(                      // interpolate spring forces at qps 
+          this->scratch_dof_values, this->scratch_qp_values);                     
+
+        for (unsigned int i = 0; i < this->scratch_qp_values.size(); ++i)         // add velocity damping
+          this->scratch_qp_values[i] = fe_values.normal_vector(i) *             // does this need to use FEFaceValues 
+            (this->scratch_qp_values[i] -  
+             this->damping_constant * m_values.get_velocity_values()[i]) *        // velocity values is at the whole cell QPs?
+            fe_values.normal_vector(i); 
+
+        std::copy(this->scratch_qp_values.begin(),                                // copy forces into output
```
Comment (drwells) [inline source/mechanics/force_contribution_lib.cc:533]: can we skip these comments? They just duplicate what is in the code.

File: source/mechanics/force_contribution_lib.cc [hunk 2]
```diff
@@ -387,6 +387,155 @@ namespace fdl
...
+            (*this->current_position)[this->scratch_cell_dofs[i]]);
+
+        extractor.get_function_values_from_local_dof_values(                      // interpolate spring forces at qps 
+          this->scratch_dof_values, this->scratch_qp_values);                     
+
+        for (unsigned int i = 0; i < this->scratch_qp_values.size(); ++i)         // add velocity damping
+          this->scratch_qp_values[i] = fe_values.normal_vector(i) *             // does this need to use FEFaceValues 
+            (this->scratch_qp_values[i] -  
+             this->damping_constant * m_values.get_velocity_values()[i]) *        // velocity values is at the whole cell QPs?
+            fe_values.normal_vector(i);
```
Comment (drwells) [inline source/mechanics/force_contribution_lib.cc:531]: These should be the deformed normal vectors provided by `MechanicsValues`, not the reference normal vectors provided by `FEValues`.

File: source/mechanics/force_contribution_lib.cc [hunk 3]
```diff
@@ -387,6 +387,155 @@ namespace fdl
...
+        dof_handler,
+        do_interpolation(dof_handler, mapping, reference_position))
+    , damping_constant(damping_constant)
+  {}
+
+  template <int dim, int spacedim, typename Number>
+  MechanicsUpdateFlags
+  PericardiumModelForce<dim, spacedim, Number>::get_mechanics_update_flags() const
+  {
+    return MechanicsUpdateFlags::update_velocity_values;
```
Comment (drwells) [inline source/mechanics/force_contribution_lib.cc:461]: We should also request the deformed normal vectors here.

**Review by drwells — CHANGES_REQUESTED** [review]
> Does this works the way I think it does where normal_vector and get_velocity_values are computed at the quadrature points on the face?

Yes.

We also need a test.

**Comment by drwells** [comment]
Take a look at `tests/mechanics/spring_01.cc`: we should just print some values and call it a day.

**Comment by marshallrdavey** [comment]
@drwells this has a test now, do you want me to write another test within the same file that has more complex movement (e.g. contraction towards the center)?

**Comment by drwells** [comment]
It's probably fine as-is.

**Review by drwells — APPROVED** [review]

**Pull request merged**